shared-data/js/helpers/getSimplestFlexDeckConfig.ts

```diff
@@ -44,29 +44,33 @@ export function getSimplestDeckConfigForProtocol(
 ): CutoutConfigProtocolSpec[] {
   // TODO(BC, 2023-11-06): abstract out the robot type
   const deckDef = getDeckDefFromRobotType(FLEX_ROBOT_TYPE)
-
   const addressableAreas =
     protocolAnalysis != null
       ? getAddressableAreasInProtocol(protocolAnalysis, deckDef)
       : []
+  // iterates through the list of required addressable areas for the protocol
   const simplestDeckConfig = addressableAreas.reduce<
     CutoutConfigProtocolSpec[]
   >((acc, addressableArea) => {
+    // finds all cutout fixtures that provide this addressable area
     const cutoutFixturesForAddressableArea = getCutoutFixturesForAddressableAreas(
       [addressableArea],
       deckDef.cutoutFixtures
     )
+    // grabs the cutout id for the addressable area
     const cutoutIdForAddressableArea = getCutoutIdForAddressableArea(
       addressableArea,
       cutoutFixturesForAddressableArea
     )
+    // grabs all possible cutout fixtures for that cutout id
     const cutoutFixturesForCutoutId =
       cutoutIdForAddressableArea != null
         ? getCutoutFixturesForCutoutId(
             cutoutIdForAddressableArea,
             deckDef.cutoutFixtures
           )
         : null
+    // finds what is currently in that specific cutout
```

Comment:
If we're going to leave my comments in, let's make this one a little more clear. Something like 
```suggestion
    // this grabs the previously found CutoutConfig if we've already added one to our acc
    // for a different addressable area required by the same CutoutId
```